inference-engine/src/vpu/common/src/ngraph/operations/static_shape_broadcast.cpp

```diff
@@ -58,7 +58,32 @@ void StaticShapeBroadcast::validate_and_infer_types() {
 
         const auto evaluatedTargetShape = ngraph::PartialShape(evaluatedDimensionValues);
         if (evaluatedTargetShape.is_static()) {
-            m_evaluatedOutputShape = evaluatedTargetShape;
+            if (m_mode.m_type == ngraph::op::BroadcastType::BIDIRECTIONAL) {
```

Comment:
You may simplify this code, since it's too hard to understand what's going on.
In order to get the final shape you should simply do:
1) select the shape with lower rank (number of dimensions) and insert ones in its beginning as this broadcasting rule is right alignment.
2) Do per-element maximum for dimensions in targetShape and inputShape
That's it.

Reply:
I've changed it, can you please take a look?

Reply:
Well, I think something like this could be much simpler:
```
auto targetShape = evaluatedDimensionValues.get_shape();
auto inputShape = get_input_partial_shape(0).get_shape();

auto& lowRankShape = targetShape.size() < inputShape.size() ? targetShape : inputShape;
auto& highRankShape = lowRankShape == targetShape ? inputShape : targetShape;

while (lowRankShape.size() < highRankShape.size()) {
    lowRankShape.insert(lowRankShape.begin(), 1);
}

for (size_t i = 0; i < targetShape.size(); i++) {
    targetShape[i] = std::max(targetShape[i], inputShape[i]);
}

m_evaluatedOutputShape = targetShape;
```

Reply:
That does look better, so I've changed it